deeplinkdispatch-processor/src/main/java/com/airbnb/deeplinkdispatch/Documentor.java

```diff
@@ -0,0 +1,129 @@
+
+    private final ProcessingEnvironment processingEnv;
+    private final Messager messager;
+
+    @Nullable private File file;
+
+    Documentor(ProcessingEnvironment processingEnv) {
+        this.processingEnv = processingEnv;
+        messager = processingEnv.getMessager();
+        initFile();
+    }
+
+    void write(List<DeepLinkAnnotatedElement> elements) {
+        if (file == null) {
+            messager.printMessage(Diagnostic.Kind.WARNING,
+                    "Output file is null, DeepLink doc not generated.");
+            return;
+        }
+        if (elements == null || elements.isEmpty()) {
+            messager.printMessage(Diagnostic.Kind.WARNING,
+                    "No deep link, DeepLink doc not generated.");
+            return;
+        }
+        PrintWriter writer;
+        try {
+            writer = new PrintWriter(new FileWriter(file), true);
+            for (DeepLinkAnnotatedElement element : elements) {
+                writer.print(element.getUri() + PROPERTY_DELIMITER);
+                String doc = formatJavaDoc(
+                        processingEnv.getElementUtils().getDocComment(element.getElement()));
+                if (doc != null) {
+                    writer.print(doc);
+                }
+                writer.print(PROPERTY_DELIMITER);
+                writer.print(element.getAnnotatedElement().getSimpleName());
+                if (element.getAnnotationType().equals(DeepLinkEntry.Type.METHOD)) {
+                    writer.print(CLASS_METHOD_NAME_DELIMITER);
+                    writer.print(element.getMethod());
+                }
+                writer.print(ELEMENT_DELIMITER);
+            }
+            writer.flush();
+            writer.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+            messager.printMessage(Diagnostic.Kind.WARNING, "DeepLink doc not generated, ");
+        }
+        messager.printMessage(Diagnostic.Kind.NOTE,
+                "DeepLink doc generated at: " + file.getPath());
+
+    }
+
+    private void initFile() {
+        String path = processingEnv.getOptions().get(DOC_OUTPUT_PROPERTY_NAME);
+        if (path != null && path.trim().length() > 0) {
+            file = new File(path);
+            if (file.isDirectory()) {
+                messager.printMessage(Diagnostic.Kind.WARNING,
+                        String.format("Specify a file path at %s to generate deep link doc.",
+                                DOC_OUTPUT_PROPERTY_NAME));
+            } else {
+                final File parentDir = file.getParentFile();
+                if (!parentDir.exists() && !parentDir.mkdirs()) {
+                    messager.printMessage(Diagnostic.Kind.WARNING,
+                            String.format("Cannot create file specified at %s.", file));
+                }
+            }
+        } else {
+            messager.printMessage(Diagnostic.Kind.WARNING,
```

Comment:
I think this could be an `INFO` since not everyone might want to generate the doc and that will cause lots of warnings